### PR TITLE
Tweak CMakeLists.txt

### PR DESCRIPTION
**Pull request by haydenroche5**
- Add generated CMake files/directories to .gitignore.
- Use lowercase for CMake commands, UPPERCASE for variables.
- Favor the CMake "option" command over SET(... CACHE BOOL ...).
- Use CMAKE_CURRENT_SOURCE_DIR in place of CMAKE_CURRENT_BINARY_DIR.
- Use CMAKE_USE_PTHREADS_INIT instead of CMAKE_HAVE_PTHREAD_H.
- Use target_include_directories on the wolfssl library target instead of include_directories.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -16,14 +16,18 @@
...
 # $ cmake .. -DCMAKE_BUILD_TYPE=Debug
 
-
-cmake_minimum_required (VERSION 2.6)
-
 ####################################################
 # Project
 ####################################################
+
+cmake_minimum_required(VERSION 2.6)
```
Comment (haydenroche5) [inline CMakeLists.txt:23]: We may want to raise this to a higher version. 2.6 is very old. I don't think it has support for some of the core modern CMake features, like targets.

File: CMakeLists.txt [hunk 2]
```diff
@@ -55,45 +59,49 @@ endif()
...
    # Use template
    configure_file(${CMAKE_CURRENT_SOURCE_DIR}/wolfssl/options.h.in 
-                  ${CMAKE_CURRENT_BINARY_DIR}/user_settings.h)
+                  ${CMAKE_CURRENT_SOURCE_DIR}/user_settings.h)
 endif()
 
 add_definitions(-DWOLFSSL_USER_SETTINGS)
 add_definitions(-DWOLFSSL_IGNORE_FILE_WARN)
-if(CMAKE_HAVE_PTHREAD_H)
+if(CMAKE_USE_PTHREADS_INIT)
```
Comment (haydenroche5) [inline CMakeLists.txt:77]: I believe this is the more up-to-date variable, based on this: https://cmake.org/cmake/help/v3.19/module/FindThreads.html

File: CMakeLists.txt (hunk 1 above)
Comment (dgarske) [reply CMakeLists.txt:23]: Go ahead and raise to an appropriate minimum version. This was not due to a requirement, just the example I used.

File: CMakeLists.txt [hunk 3]
```diff
@@ -55,45 +59,49 @@ endif()
...
 endif()
 
 ####################################################
-# Source Files
+# Library Target and Source Files
 ####################################################
-include_directories(${CMAKE_CURRENT_SOURCE_DIR}/.)
-include_directories(${CMAKE_CURRENT_BINARY_DIR}/.)
 
 file(GLOB LIB_SOURCE_FILES
```
Comment (haydenroche5) [inline CMakeLists.txt:85]: In a future commit, I would propose we stop globbing source files for the reasons outlined here: https://stackoverflow.com/questions/1027247

File: CMakeLists.txt (hunk 2 above)
Comment (dgarske) [reply CMakeLists.txt:77]: Thanks

File: CMakeLists.txt (hunk 3 above)
Comment (dgarske) [reply CMakeLists.txt:85]: That's fine with me. It's a bit of additional work when new files are added, but that is okay.

File: CMakeLists.txt [hunk 4]
```diff
@@ -55,45 +59,49 @@ endif()
...
+add_library(wolfssl ${LIB_SOURCE_FILES})
 
 ####################################################
-# Output Files
+# Include Directories
 ####################################################
 
-# Build wolfssl library
-add_library(wolfssl ${LIB_SOURCE_FILES})
+target_include_directories(wolfssl
```
Comment (haydenroche5) [inline CMakeLists.txt:95]: The advantage to target_include_directories over include_directories is that the former is target-scoped, and the latter is directory-scoped. This change makes it so the include directories specified here are limited to the wolfssl target/library, as opposed to applying to all targets in this file. For more: https://stackoverflow.com/questions/31969547

Comment (dgarske) [reply CMakeLists.txt:95]: Thanks

**Review by dgarske — CHANGES_REQUESTED** [review]
Awesome feedback. Go ahead and implement your suggestions.

File: CMakeLists.txt [hunk 5]
```diff
@@ -153,15 +239,17 @@ if(BUILD_TESTS)
...
-    set_target_properties( unit_test PROPERTIES COMPILE_FLAGS "-DNO_MAIN_DRIVER" )
+        tests/api.c
+        tests/hash.c
+        tests/srp.c
+        tests/suites.c
+        tests/unit.c
+        examples/server/server.c
+        examples/client/client.c
+    )
+    target_compile_options(unit_test PUBLIC "-DNO_MAIN_DRIVER")
```
Comment (haydenroche5) [inline CMakeLists.txt:252]: This has the same effect as the old code, it's just a bit cleaner.

**Review by dgarske — APPROVED** [review]

**Pull request merged**